plugin/src/main/java/com/denizenscript/denizen/objects/EntityTag.java

```diff
@@ -2936,6 +2936,176 @@ else if (object.getBukkitEntity() instanceof Hanging hanging) {
+                    return null;
+                }
+                Entity entity = param.getBukkitEntity();
+                if (entity == null) {
+                    attribute.echoError("Invalid entity '" + param.debuggable() + "<W>' specified: must be spawned.");
+                    return null;
+                }
+                return new ElementTag(warden.getAnger(entity));
+            });
+
+            // <--[tag]
+            // @attribute <EntityTag.anger_level>
+            // @returns ElementTag
+            // @description
+            // Returns a warden's current anger level, which will be any of <@link url https://hub.spigotmc.org/javadocs/spigot/org/bukkit/entity/Warden.AngerLevel.html>.
+            // -->
+            tagProcessor.registerTag(ElementTag.class, "anger_level", (attribute, object) -> {
+                if (!(object.getBukkitEntity() instanceof Warden warden)) {
+                    return null;
+                }
+                return new ElementTag(warden.getAngerLevel());
+            });
+
+            // <--[tag]
+            // @attribute <EntityTag.angry_at>
+            // @returns EntityTag
+            // @description
+            // Returns the entity a warden is the most angry at, if any.
+            // -->
+            tagProcessor.registerTag(EntityFormObject.class, "angry_at", (attribute, object) -> {
+                if (!(object.getBukkitEntity() instanceof Warden warden)) {
+                    return null;
+                }
+                Entity angryAt = warden.getEntityAngryAt();
+                return angryAt != null ? new EntityTag(angryAt).getDenizenObject() : null;
+            });
+
+            // <--[mechanism]
+            // @object EntityTag
+            // @name clear_anger
+            // @input EntityTag
+            // @description
+            // Clears a warden's anger towards the input entity.
+            // @tags
+            // <EntityTag.anger_at[<entity>]>
+            // -->
+            tagProcessor.registerMechanism("clear_anger", false, EntityTag.class, (object, mechanism, input) -> {
+                if (!(object.getBukkitEntity() instanceof Warden warden)) {
+                    mechanism.echoError("Cannot adjust '" + object.debuggable() + "<W>': must be a warden.");
+                    return;
+                }
+                Entity entity = input.getBukkitEntity();
+                if (entity == null) {
+                    mechanism.echoError("Invalid entity '" + input.debuggable() + "<W>' specified: must be spawned.");
+                    return;
+                }
+                warden.clearAnger(entity);
+            });
+
+            // <--[mechanism]
+            // @object EntityTag
+            // @name anger_at
+            // @input MapTag
+            // @description
+            // Sets a warden's anger towards a specific entity.
+            // The input map needs to have the following keys:

```

Comment:
These mechs could really use some `@examples` to make sure usage is clear

Reply:
Added examples to tags & mechs